DataDog/serverless-plugin-datadog
# support subdomain in output links

### What does this PR do?

Allows customization of the subdomain used in output urls.

**Before (and default if subdomain not set)**
```
Datadog Monitoring
functions
  sync: https://app.datadoghq.com/functions/myservice-sync:us-east-1:123456789123:aws?source=sls-plugin

...

View Serverless Monitors
  https://app.datadoghq.com/monitors/manage?q=tag%3A%28%22env%3Aprod%22AND%22service%3Amyservice%22%29
```

**After (with subdomain set to custom)**
```
Datadog Monitoring
functions
  sync: https://custom.datadoghq.com/functions/myservice-sync:us-east-1:123456789123:aws?source=sls-plugin

...

View Serverless Monitors
  https://custom.datadoghq.com/monitors/manage?q=tag%3A%28%22env%3Aprod%22AND%22service%3Amyservice%22%29
```

### Motivation

Makes clicking the links more convenient if you interact with multiple orgs.

### Testing Guidelines

- With no changes to your code view output of serverless to see the usual app.datadoghq links
- Set subdomain variable to a value and view output of serverless to see urls output using this subdomain 

### Additional Notes

This feature is available in datadog-ci synthetics runs and is inspired by that work

### Types of changes


### Check all that apply
